app/lib/core/Search/SearchIndexer.php

```diff
@@ -1437,7 +1408,7 @@ private function _indexAttribute($pt_subject, $pn_row_id, $pm_element_code_or_id
 
 						$va_children_ids = $pt_subject->getHierarchyAsList($pn_row_id, array('idsOnly' => true, 'includeSelf' => false));
 
-						if (!$pb_reindex_mode && is_array($va_children_ids) && sizeof($va_children_ids) > 0) {
```

Comment:
`$pb_reindex_mode` was undefined, which is falsy, so `!$pb_reindex_mode` is always going to be true, therefore this check can be removed.  Good pick up though :-)